src/workers/postUpdated.ts

```diff
@@ -294,19 +301,27 @@ const updatePost = async ({
   }
 
   const title = data?.title || databasePost.title;
-  const updateBecameVisible =
-    content_type === PostType.Freeform
-      ? databasePost.visible
-      : !databasePost.visible && !!title?.length;
 
   data.id = databasePost.id;
   data.title = title;
-  data.visible = updateBecameVisible;
-  data.visibleAt = updateBecameVisible
-    ? databasePost.visibleAt ?? data.metadataChangedAt
-    : null;
   data.sourceId = data.sourceId || databasePost.sourceId;
 
+  let updateBecameVisible = false;
+
+  if (!databasePost.visible) {
+    updateBecameVisible = getPostVisible({ post: { title } });
+  }
+
+  if (updateBecameVisible) {
+    data.visible = true;
+  } else {
+    data.visible = databasePost.visible;
+  }
+
+  if (data.visible && !databasePost.visibleAt) {
+    data.visibleAt = data.metadataChangedAt;
+  }
```

Comment:
if post is visible but missing `visibleAt` set it here. This fixes current issue in production where we have a bunch of posts that are `visible: true` but `visibleAt: null`.

Check:
```
select count(*) from post p where p.visible = true and p."visibleAt" is null
```